posthog/hogql_queries/web_analytics/web_overview.py

```diff
@@ -22,17 +25,56 @@ class WebOverviewQueryRunner(WebAnalyticsQueryRunner):
     response: WebOverviewQueryResponse
     cached_response: CachedWebOverviewQueryResponse
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.preaggregated_query_builder = WebOverviewPreAggregatedQueryBuilder(self)
+
     def to_query(self) -> ast.SelectQuery:
         return self.outer_select
 
-    def calculate(self):
-        response = execute_hogql_query(
-            query_type="overview_stats_pages_query",
-            query=self.to_query(),
-            team=self.team,
-            timings=self.timings,
-            modifiers=self.modifiers,
-            limit_context=self.limit_context,
+    def get_pre_aggregated_response(self):
+        should_use_preaggregated = (
+            self.modifiers
+            and self.modifiers.useWebAnalyticsPreAggregatedTables
+            and self.preaggregated_query_builder.can_use_preaggregated_tables()
+        )
+
+        if not should_use_preaggregated:
+            return None
+
+        try:
+            response = execute_hogql_query(
+                query_type="web_overview_preaggregated_query",
+                query=self.preaggregated_query_builder.get_query(),
+                team=self.team,
+                timings=self.timings,
+                modifiers=self.modifiers,
+                limit_context=self.limit_context,
+            )
+
+            # We could have a empty result in normal conditions but also when we're recreating the tables.
+            # While we're testing, if it is a empty result, let's  fallback on using the
+            # regular queries to be extra careful.
+            assert response.results
+
+            return response
+        except Exception:
+            return None
+
+    def calculate(self) -> WebOverviewQueryResponse:
+        pre_aggregated_response = self.get_pre_aggregated_response()
+
+        response = (
```

Comment:
This would be slightly more readable with an actual if/else and then assigning `response` inside the 2 bodies